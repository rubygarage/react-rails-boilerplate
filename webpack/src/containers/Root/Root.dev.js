import PropTypes from 'prop-types';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { renderRoutes } from 'react-router-config';
import IntlProviderRedux from 'containers/IntlProviderRedux';
import MemoizedBrowserHistory from 'helpers/memoizedBrowserHistory';
import createMemoryHistory from 'history/createMemoryHistory';
import routes from 'routes';
import styles from 'assets/styles/global.css' // eslint-disable-line

export default class Root extends React.PureComponent {
  render() {
    const { store, type, req } = this.props;
    let history = {};
    type === 'server' ? history = createMemoryHistory() : history = new MemoizedBrowserHistory();

    if (req) { history.push(req.url); }

    return (
      <Provider store={store}>
        <IntlProviderRedux>
          <Router history={history} >
            {renderRoutes(routes)}
          </Router>
        </IntlProviderRedux>
      </Provider>
    );
  }
}

Root.propTypes = {
  store: PropTypes.shape.isRequired,
  // history: PropTypes.object.isRequired,
  // routes: PropTypes.node.isRequired,
  type: PropTypes.string,
  // renderProps: PropTypes.object,
  req: PropTypes.shape.isRequired,
};
