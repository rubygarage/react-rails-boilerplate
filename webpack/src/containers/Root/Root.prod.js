import PropTypes from 'prop-types';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { renderRoutes } from 'react-router-config';
import IntlProviderRedux from 'containers/IntlProviderRedux';
import MemoizedBrowserHistory from 'helpers/memoizedBrowserHistory';
import createMemoryHistory from 'history/createMemoryHistory';
import routes from 'routes';
import styles from 'assets/styles/global.css' // eslint-disable-line
import GoogleAnalytics from 'utils/googleAnalytics';

export default class Root extends React.PureComponent {
  render() {
    const { store, type, req } = this.props;
    let history = {};
    type === 'server' ? history = createMemoryHistory() : history = new MemoizedBrowserHistory();

    const googleAnalytics = new GoogleAnalytics();
    history.listen(() => {
      googleAnalytics.triggerPageView();
    });

    if (req) { history.push(req.url); }

    return (
      <Provider store={store}>
        <IntlProviderRedux>
          <Router history={history} >
            {renderRoutes(routes)}
          </Router>
        </IntlProviderRedux>
      </Provider>
    );
  }
}

Root.propTypes = {
  store: PropTypes.shape({}),
  type: PropTypes.string,
  req: PropTypes.shape({}),
};
