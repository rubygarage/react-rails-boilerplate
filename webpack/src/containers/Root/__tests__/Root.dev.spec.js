import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Root from 'containers/Root/Root.dev';

describe('Root.dev container', () => {
  const store = configureStore()({});
  store.dispatch = jest.fn();
  const defaultProps = {
    store,
  };

  const container = shallow(<Root {...defaultProps} />);

  it('renders Root', () => {
    expect(container).toMatchSnapshot();
  });
});
