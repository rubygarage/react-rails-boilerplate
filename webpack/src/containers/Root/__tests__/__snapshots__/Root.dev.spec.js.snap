// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root.dev container renders Root 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(IntlProvider)>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Switch>
        <Route
          exact={true}
          key="0"
          path="/"
          render={[Function]}
        />
        <Route
          exact={true}
          key="1"
          path="/sign_in"
          render={[Function]}
        />
        <Route
          exact={true}
          key="2"
          path="/sign_up"
          render={[Function]}
        />
        <Route
          key="3"
          path="/sign_up/success"
          render={[Function]}
        />
        <Route
          key="4"
          path="/email_confirmation/:confirmation_token"
          render={[Function]}
        />
        <Route
          exact={true}
          key="5"
          path="/reset_password"
          render={[Function]}
        />
        <Route
          exact={true}
          key="6"
          path="/reset_password/:reset_password_token"
          render={[Function]}
        />
        <Route
          key="7"
          path="/user/:id"
          render={[Function]}
        />
        <Route
          key="8"
          path="/*"
          render={[Function]}
        />
      </Switch>
    </Router>
  </Connect(IntlProvider)>
</Provider>
`;
